localization/pose_instability_detector/src/pose_instability_detector.cpp

```diff
@@ -0,0 +1,185 @@
+    const double time_diff_sec = time_diff.seconds();
+    if (time_diff_sec < 0.0) {
+      RCLCPP_WARN_STREAM(get_logger(), "time_diff_sec(" << time_diff_sec << ") < 0.0");
+      continue;
+    }
+
+    // quat to rpy
+    auto [ang_x, ang_y, ang_z] = quat_to_rpy(pose.orientation);
+
+    // rpy update
+    ang_x += twist.angular.x * time_diff_sec;
+    ang_y += twist.angular.y * time_diff_sec;
+    ang_z += twist.angular.z * time_diff_sec;
+    tf2::Quaternion quat;
+    quat.setRPY(ang_x, ang_y, ang_z);
+
+    // Convert twist to world frame
+    tf2::Vector3 linear_velocity(twist.linear.x, twist.linear.y, twist.linear.z);
+    linear_velocity = tf2::quatRotate(quat, linear_velocity);
+
+    // update
+    pose.position.x += linear_velocity.x() * time_diff_sec;
+    pose.position.y += linear_velocity.y() * time_diff_sec;
+    pose.position.z += linear_velocity.z() * time_diff_sec;
+    pose.orientation.x = quat.x();
+    pose.orientation.y = quat.y();
+    pose.orientation.z = quat.z();
+    pose.orientation.w = quat.w();
+    prev_time = curr_time;
+  }
+
+  // compare pose and latest_odometry_
+  const Pose latest_ekf_pose = latest_odometry_.pose.pose;
+  PoseStamped ekf_to_odom;
+  ekf_to_odom.pose = tier4_autoware_utils::inverseTransformPose(pose, latest_ekf_pose);
+  ekf_to_odom.header.stamp = latest_odometry_.header.stamp;
+  ekf_to_odom.header.frame_id = "base_link";
+  const geometry_msgs::msg::Point pos = ekf_to_odom.pose.position;
+  const auto [ang_x, ang_y, ang_z] = quat_to_rpy(ekf_to_odom.pose.orientation);
+  const std::vector<double> values = {pos.x, pos.y, pos.z, ang_x, ang_y, ang_z};
+
+  // publish debug
+  const rclcpp::Time stamp = latest_odometry_.header.stamp;
+  diff_position_x_pub_->publish(tier4_debug_msgs::build<Float64Stamped>().stamp(stamp).data(pos.x));
+  diff_position_y_pub_->publish(tier4_debug_msgs::build<Float64Stamped>().stamp(stamp).data(pos.y));
+  diff_position_z_pub_->publish(tier4_debug_msgs::build<Float64Stamped>().stamp(stamp).data(pos.z));
+  diff_angle_x_pub_->publish(tier4_debug_msgs::build<Float64Stamped>().stamp(stamp).data(ang_x));
+  diff_angle_y_pub_->publish(tier4_debug_msgs::build<Float64Stamped>().stamp(stamp).data(ang_y));
+  diff_angle_z_pub_->publish(tier4_debug_msgs::build<Float64Stamped>().stamp(stamp).data(ang_z));
+
+  const std::vector<double> thresholds = {threshold_diff_position_x_, threshold_diff_position_y_,
+                                          threshold_diff_position_z_, threshold_diff_angle_x_,
+                                          threshold_diff_angle_y_,    threshold_diff_angle_z_};
+
+  const std::vector<std::string> labels = {"diff_position_x", "diff_position_y", "diff_position_z",
+                                           "diff_angle_x",    "diff_angle_y",    "diff_angle_z"};
+
+  // publish diagnostics
+  DiagnosticArray diagnostics;
+  diagnostics.header.stamp = stamp;
+
+  DiagnosticStatus status;
+  status.name = "pose_instability_detector";
```

Comment:
I'd like to include "localization: " as a prefix to make the name of the diagnostic.
```suggestion
  status.name = "localization: pose_instability_detector";
```